Lib/unittest/mock.py

```diff
@@ -1282,6 +1282,8 @@ def __init__(
                 f'Cannot spec attr {attribute!r} as the spec_set '
                 f'target has already been mocked out. [spec_set={spec_set!r}]')
 
+        if unsafe:
+            kwargs.update({'unsafe': unsafe})
```

Comment:
why not

```python
kwargs['unsafe'] = unsafe
```

?

Reply:
Hi @gst,
This was done to emulate how passing unsafe via kwargs was originally designed.
That is, if unsafe is not explicitly set, kwargs don't default unsafe to False, it's just not present in kwargs. 
There are also several test cases already written where, if unsafe is not set then kwargs is expected not to contain the unsafe key.
Particularly in this bug fix, I didn't want to introduce any behavioral changes.

Reply:
well I think both of us got their points actually :)

```
def __init__(
            self, getter, attribute, new, spec, create,
            spec_set, autospec, new_callable, kwargs, *, unsafe=False
        ):
```

given there is none annotation to the `kwargs` we can, eventually, do anything with it.
might be it seems to be a regular dict/mapping though.

and using regular `dict[index] = value` looks the way to do it ?

Reply:
I agree, the problem as you are pointing out is in the initializer where unsafe is pulled out and made into a named argument.
My original thought was to just change the initializer and remove that unsafe from it. That is probably the best solution and how it was originally.


But then I realized all the public/exposed APIs like for example:
```
def patch(
        target, new=DEFAULT, spec=None, create=False,
        spec_set=None, autospec=None, new_callable=None, *, unsafe=False, **kwargs
):
def _patch_object(
        target, attribute, new=DEFAULT, spec=None,
        create=False, spec_set=None, autospec=None,
        new_callable=None, *, unsafe=False, **kwargs
    ):
```
have unsafe also made into an independent named argument. Since it is these public/exposed APIs that get called **kwargs at this point is always a dict and that get passed internally everywhere in the mock module.

It is strange why kwargs is not checking for None in the _patch initializer(so one can test _patch class independently), and I see that some initializer calls go into great extend to ensure that an empty dict is passed into the initializer. Like the following piece of code that can be found in the _patch_multiple function.
```
this_patcher = _patch(
    getter, attribute, new, spec, create, spec_set,
    autospec, new_callable, {}
)
```

So as it is right now, the initializer for _patch class is some what coupled with the calling public API in the sense that whoever calls it has to always pass a dict for kwargs. It's a little unsafe but could just have been the design from the beginning.

So if we always get a dict for kwargs then why not use,
`kwargs['unsafe'] = unsafe` instead of `kwargs.update({'unsafe': unsafe})`?
You are right, it's probably more efficient to use `dict[index] = value` style here.
However the `if` conditional statement that checks if unsafe is set to True will need to stay. Otherwise we will change the behavior of kwargs and that break some testcases written in other modules.
Would that be okay?